### PR TITLE
ADC additions and git hooks

### PR DESCRIPTION
**Pull request by johnandersen777**
The ADC isn't reading correctly, so the assert_failure for it has been gotten rid of in main.
This is still very rough around the edges at this point

### CODE REVIEW

**Comment by johnandersen777** [comment]
Rebased onto master :+1:

**Review by johnandersen777 — COMMENTED** [review]
Test comment

File: src/adc.c [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+ * Returns
+ *  ADC_OK
+ *    Valid pin has been selected for conversion
+ *  ADC_INVALID_CONVERT_PIN
+ *    Caller passed in an invalid pin
+ */
+adc_status_t adc_select_conversion_pin(adc_convert_t pin_to_convert) {
+  switch (pin_to_convert) {
+    case ADC_CONVERT_PA1:
+      ADC1->CHSELR |= ADC_CHSELR_CHSEL1;
```
Comment (johnandersen777) [inline src/adc.c:88]: I think this is wrong

File: src/main.c [hunk 2]
```diff
@@ -66,13 +61,43 @@ int main(void)
...
+    assert_failed(__FILE__, __LINE__);
+  }
+
+  __enable_irq();
+
 	while (1) {
 		GPIOB->ODR |= GPIO_MODER_MODER1_1;
 		delay(500000);
+    /* TODO Fix adc_conversion, currently this returns a timeout error */
+    adc_status = adc_convert(ADC_CONVERT_PA1);
```
Comment (johnandersen777) [inline src/main.c]: I'm not sure if this is the right pin or where this is physically on the breakout board.

**Review by fouric — APPROVED** [review]

**Pull request merged**